rmgpy/rmg/main.py

```diff
@@ -1417,21 +1468,121 @@ def generateCanteraFiles(self, chemkinFile, **kwargs):
+                            break
+                    else:
+                        raise RuntimeError('Species {0} was defined in the restart file, but was not included in the'
+                                           'core.'.format(spc))
+
+                # Append the remaining species left in the core to the very end
+                self.reactionModel.core.species = reordered_core_species + self.reactionModel.core.species
+
+                # If we are restarting we must be able to handle all four possible combinations of whether or not
+                # filtering was used in the restart job or if filtering is used in the current job. To summarize:
+                #
+                # If filtering is being used in this job, we only need to set the threshold flags properly because the
+                # react flags are updated based on this before the first enlarge. Otherwise, we need to explicitly set
+                # the react flags properly because the thresholds will not be used.
+                #
+                # If filtering was used in the job we are restarting from, we can load these values in for all of the
+                # species present in that job to help us set the current threshold or react flags.
+                # Otherwise, if filtering was not used in the previous job then we can assume that all species
+                # present in the job we are restarting from have already reacted, and thus either set their
+                # filters to be True or their react flags to be false.
+
+                # Fill in the filter tensors.
+                if filters_found:  # Add in the filter data where we have it.
+                    # Note that additional species that have been defined in the input file but were not
+                    # present in the restart core seed will be reacted.
+
+                    if self.filterReactions:  # Filling in the filter thresholds will suffice
+                        # Fill in the newly initialized filter tensors
+                        self.unimolecularThreshold = unimolecularThreshold
+                        self.bimolecularThreshold = bimolecularThreshold
+                        if self.trimolecular:
+                            self.trimolecularThreshold = trimolecularThreshold
+
+                    else:  # We must set the react flags instead. If it was `True` in the threshold, it should not react
+                        self.unimolecularReact = np.logical_not(unimolecularThreshold)
+                        self.bimolecularReact = np.logical_not(bimolecularThreshold)
+                        if self.trimolecular:
+                            self.trimolecularReact = np.logical_not(trimolecularThreshold)
+
+                else:  # Assume that all species found in the restart core seed have already been reacted
+                    ind = len(restartSpeciesList)  # Index where the additional species begin
```

Comment:
Is this the same as `numRestartSpecies`?

Reply:
Yes, but it had been a while before I had used this, and `ind` here was shorter for this purpose, but did not make sense when I needed it earlier. What would you prefer?